#ifndef __ANOSYS_HPP__
#define __ANOSYS_HPP__
#include <yaml-cpp/yaml.h>

#include "AnoPTv8.hpp"
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/*
参数功能示例
1、先定义一个参数，注意参数类型
2、定义参数信息，依次为参数指针，参数类型，最小值，最大值，参数名称，参数说明
3、注册参数信息
*/
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
void AnoPTv8ParInit(void);
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/*
命令功能示例
*/
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
void AnoPTv8CmdInit(void);
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/*
 */
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
void readSettings(void);
std::string getYamlSet_str(const YAML::Node& node, const std::string& key, const std::string _default);
double getYamlSet_double(const YAML::Node& node, const std::string& key, const double _default);
std::string toHex(const uint8_t* data, const uint16_t len);
#endif